src/Bicep.Core/Registry/OciModuleRegistry.cs

```diff
@@ -129,10 +131,10 @@ public override bool TryGetLocalModuleEntryPointUri(OciArtifactModuleReference r
                 || string.IsNullOrWhiteSpace(documentationUri))
             {
                 // Automatically generate a help URI for public MCR modules
-                if (ociArtifactModuleReference.Registry == LanguageConstants.McrRegistry && ociArtifactModuleReference.Repository.StartsWith(LanguageConstants.McrRepositoryPrefix, StringComparison.Ordinal))
+                if (ociArtifactModuleReference.Registry == BicepPublicMcrRegistry && ociArtifactModuleReference.Repository.StartsWith(LanguageConstants.McrRepositoryPrefix, StringComparison.Ordinal))
                 {
-                    var repository = ociArtifactModuleReference.Repository.Substring(LanguageConstants.McrRepositoryPrefix.Length);
-                    return $"https://github.com/Azure/bicep-registry-modules/tree/{repository}/{ociArtifactModuleReference.Tag}/modules/{repository}/README.md";
+                    var moduleName = ociArtifactModuleReference.Repository.Substring(LanguageConstants.McrRepositoryPrefix.Length);
+                    return ociArtifactModuleReference.Tag is null ? null : GetPublicBicepModuleDocumentationUri(moduleName, ociArtifactModuleReference.Tag);
```

Comment:
nit: It might be worth handling null tag in `GetPublicBicepModuleDocumentationUri` to avoid duplicating the ternary operator in the calling methods.

Reply:
I'm hesitant to do that in this case because GetPublicBicepModuleDocumentationUri could theoretically return a URI for version=null (i.e. return the current readme.md URI rather than one associated with a specific version), but in this scenario, I'm not expecting a null tag and I'd rather leave the behavior undefined.